### PR TITLE
Remove need for context in Sentry.init for Android 

### PR DESCRIPTION
**Pull request by buenaflor**
Ref #114 

Eliminates the need for context in init - context can be passed in through a simple `ContextProvider`.

### CODE REVIEW

File: sentry-kotlin-multiplatform/src/androidMain/AndroidManifest.xml [hunk 1]
```diff
@@ -8,7 +8,14 @@
             android:name="io.sentry.android.core.SentryInitProvider"
             android:authorities="${applicationId}.SentryInitProvider"
             android:exported="false"
-            tools:node="remove"/>
+            tools:node="remove">
+        </provider>
+        <provider
+            android:name="io.sentry.kotlin.multiplatform.ContextProvider"
+            android:authorities="${applicationId}.ContextProvider"
```
Comment (markushi) [inline sentry-kotlin-multiplatform/src/androidMain/AndroidManifest.xml:15]: Maybe makes sense to change this to `SentryContextProvider`, to reduce the risk of clashing with any user-defined app authorities.

```suggestion
            android:authorities="${applicationId}.SentryContextProvider"
```

File: sentry-kotlin-multiplatform/src/androidUnitTest/kotlin/io/sentry/kotlin/multiplatform/BaseSentryTest.kt [hunk 2]
```diff
@@ -1,15 +1,22 @@
...
     actual fun sentryInit(optionsConfiguration: OptionsConfiguration) {
-        val context = InstrumentationRegistry.getInstrumentation().targetContext
-        Sentry.init(context, optionsConfiguration)
+        Sentry.init(optionsConfiguration)
+    }
+
+    @BeforeTest
+    open fun setUp() {
+        // Set up the provider needed for Sentry.init on Android
+        val provider = Robolectric.buildContentProvider(ContextProvider::class.java)
```
Comment (markushi) [inline sentry-kotlin-multiplatform/src/androidUnitTest/kotlin/io/sentry/kotlin/multiplatform/BaseSentryTest.kt:19]: Didn't know that one, nice!

File: sentry-kotlin-multiplatform/src/androidUnitTest/kotlin/io/sentry/kotlin/multiplatform/ContextProviderTest.kt [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import android.content.ContentValues
+import android.net.Uri
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.mock
+
+@RunWith(AndroidJUnit4::class)
+class ContextProviderTest : BaseSentryTest() {
```
Comment (markushi) [inline sentry-kotlin-multiplatform/src/androidUnitTest/kotlin/io/sentry/kotlin/multiplatform/ContextProviderTest.kt:12]: `l`: Maybe add one test to verify the `context` is non-null after `onCreate` is called?

File: sentry-kotlin-multiplatform/src/androidMain/kotlin/io/sentry/kotlin/multiplatform/SentryInit.android.kt [hunk 4]
```diff
@@ -1,15 +1,72 @@
...
+/**
+ * A ContentProvider that does NOT store or provide any data for read or write operations.
+ *
+ * It's only purpose is to retrieve and store the application context in an internal top-level
+ * variable [applicationContext]. The context is used for [SentryAndroid.init].
+ *
+ * This does not allow for overriding the abstract query, insert, update, and delete operations
+ * of the [ContentProvider].
+ */
+internal class ContextProvider : ContentProvider() {
```
Comment (markushi) [inline sentry-kotlin-multiplatform/src/androidMain/kotlin/io/sentry/kotlin/multiplatform/SentryInit.android.kt:31]: I think the default proguard rules already exclude any `ContentProvider` classes from being obfuscated these days, but maybe worth double checking if the name is kept in a release build.

**Review by markushi — APPROVED** [review]
Nice one, LGTM!

File: sentry-kotlin-multiplatform/src/androidMain/kotlin/io/sentry/kotlin/multiplatform/SentryInit.android.kt (hunk 4 above)
Comment (buenaflor) [reply sentry-kotlin-multiplatform/src/androidMain/kotlin/io/sentry/kotlin/multiplatform/SentryInit.android.kt]: Yup, it's no obfuscated in the release build
